reporting/miro_transformer/src/transform.py

```diff
@@ -0,0 +1,83 @@
+import json
+from datetime import datetime
+
+
+def transform(record):
+    """
+    Parameters
+    ----------
+    record : dict
+        the raw data from VHS in a malleable dict format
+
+    Returns
+    -------
+    transformed_record : dict
+        record with necessary transformations applied, ready to be formatted for
+        elasticsearch ingestion
+    """
+    original_data = json.loads(record["data"])
+    transformed = drop_redundant_fields(original_data, keys_to_drop)
+    transformed = clean_dates(transformed)
+    return transformed
+
+
+def drop_redundant_fields(original_data, keys_to_drop):
+    """
+    strip out any of the original VHS fields which are either empty, sparse,
+    redundantly filled, or just generally unwanted in our ES index
+    """
+    clean_data = {
+        key: value for key, value in original_data.items() if key not in keys_to_drop
```

Comment:
Ooh, be careful here.

Your `keys_to_drop` is a list, and list lookups are O(n) – Python has to check every element of the list to see if it compares.

Then this check is O(k * n), where k is the length of `original_data`.

Since you don't care about order or duplicate elements, `keys_to_drop` could be a set instead, and set lookups are O(1) – it uses hashing to identify elements, so lookups are faster.

Make your list a set instead!